src/openzaak/utils/query.py

```diff
@@ -23,3 +36,113 @@ def delete(self, *args, **kwargs):
 
     # see django.db.models.query.QuerySet.delete
     delete.queryset_only = True
+
+
+class LooseFkAuthorizationsFilterMixin:
+    auth_fields = []
+    loose_fk_field = None
+    vertrouwelijkheidaanduiding_use = True
+    authorizations_lookup = None
+
+    @property
+    def prefix(self):
+        return (
+            "" if not self.authorizations_lookup else f"{self.authorizations_lookup}__"
+        )
+
+    def get_loose_fk_object(
+        self, authorization, local=True
+    ) -> Union[models.Model, str]:
+        loose_fk_url = getattr(authorization, self.loose_fk_field)
+        if local:
+            loose_fk_object_path = urlparse(loose_fk_url).path
+            loose_fk_object = get_resource_for_path(loose_fk_object_path)
+        else:
+            loose_fk_object = loose_fk_url
+        return loose_fk_object
+
+    def build_queryset(self, filters) -> models.QuerySet:
+        if self.vertrouwelijkheidaanduiding_use:
+            # annotate the queryset so we can map a string value to a logical number
+            order_case = VertrouwelijkheidsAanduiding.get_order_expression(
+                f"{self.prefix}vertrouwelijkheidaanduiding"
+            )
+            annotations = {"_va_order": order_case}
+            # bring it all together now to build the resulting queryset
+            queryset = self.annotate(**annotations).filter(**filters)
+
+        else:
+            del filters["_va_order__lte"]
+            queryset = self.filter(**filters)
+        return queryset
+
+    def get_filters(self, scope, authorizations, local=True) -> dict:
+        prefix = self.prefix
+        loose_fk_field = (
+            f"_{self.loose_fk_field}" if local else f"_{self.loose_fk_field}_url"
```

Comment:
Works and is a pragmatic approach, but as soon as we change the naming scheme, this breaks. Something to keep in mind :)